avocado/core/loader.py

```diff
@@ -708,16 +711,19 @@ def _make_avocado_tests(self, test_path, make_broken, subtests_filter,
             tests = self._find_avocado_tests(test_path)
             if tests:
                 test_factories = []
-                for test_class, info in tests.items():
+                for test_class in tests:
```

Comment:
well `iteritems()` would still be better here to avoid the need to look for `tests[test_class]` inside the branches... Alternatively you can get the path once on line 716 before the for loop.